cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java

```diff
@@ -189,4 +195,30 @@ public boolean isShutdown() {
             throw new RejectedExecutionException("Database is closed");
         }
     }
+
+    /**
+     * Returns the SQLite Version.
+     * @return The SQLite version or "Unknown" if the version could not be determined.
+     */
+    public String getSQLiteVersion() {
```

Comment:
We could do that, but it would change the start up profile of the datastore, since currently we just queue operations for migrations etc. It may cause ANRs in android for example if the datastore is created on the UI thread. I'd expect other calls into the queue via the datastore eg `changes` to be on a background thread.  

We could get around that by not waiting on the version number to be returned, and by using an in memory database to get the version, but we'd have to be careful that on older devices that we still support because of their android version that we don't cause ANRs. 

We'd also then have to set the thread name via the SQLCallable, since we couldn't wait for the name to be returned. It would mean that the SQLCallable would become coupled to the Queue, which I think is the wrong way to go.

Instead of asking for the version every time, what we could do is cache the result in a field so if it's not `null` we return straight away otherwise wait for the task to be processed on the queue.